### ISSUE TITLE
Exception on sync with filter

### ISSUE THREAD

**Issue by yo8192** [vsespb/mt-aws-glacier#63]
Hi,

I'm getting an exception when trying a sync with a filter:
(please note I've shortened the file names in the outputs below)

 $ ./mtglacier.pl sync  --config=glacier.cfg --dir=. --new --filter '+backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_\* -**'
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
    LANGUAGE = "en_GB:en",
    LC_ALL = (unset),
    LC_MESSAGES = "fr_FR.UTF-8",
    LC_COLLATE = "fr_FR.UTF-8",
    LC_CTYPE = "fr_FR.UTF-8",
    LANG = "en_GB.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
MT-AWS-Glacier, Copyright 2012-2013 Victor Efimov http://mt-aws.com/ Version 1.103

PID 4234 Started worker
PID 4235 Started worker
PID 4236 Started worker
PID 4237 Started worker
UNEXPECTED ERROR (parent): file not exists at lib/App/MtAws/Utils.pm line 160
    App::MtAws::Utils::file_size('backup-pictures_...') called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 50
    App::MtAws::QueueJob::MultipartCreate::init_file('App::MtAws::QueueJob::MultipartCreate=HASH(0x16c93e8)') called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 72
    App::MtAws::QueueJob::MultipartCreate::on_create('App::MtAws::QueueJob::MultipartCreate=HASH(0x16c93e8)') called at lib/App/MtAws/QueueJob.pm line 99
    App::MtAws::QueueJob::next('App::MtAws::QueueJob::MultipartCreate=HASH(0x16c93e8)') called at lib/App/MtAws/QueueJob.pm line 88
    App::MtAws::QueueJob::next('App::MtAws::QueueJob::UploadMultipart=HASH(0x16c9a30)') called at lib/App/MtAws/QueueJob.pm line 88
    App::MtAws::QueueJob::next('App::MtAws::QueueJob::Upload=HASH(0xc75b58)') called at lib/App/MtAws/QueueJob/Iterator.pm line 61
    App::MtAws::QueueJob::Iterator::find_next_job('App::MtAws::QueueJob::Iterator=HASH(0xeceb30)') called at lib/App/MtAws/QueueJob/Iterator.pm line 100
    App::MtAws::QueueJob::Iterator::on_itt_and_jobs('App::MtAws::QueueJob::Iterator=HASH(0xeceb30)') called at lib/App/MtAws/QueueJob.pm line 99
    App::MtAws::QueueJob::next('App::MtAws::QueueJob::Iterator=HASH(0xeceb30)') called at lib/App/MtAws/QueueJob/Iterator.pm line 61
    App::MtAws::QueueJob::Iterator::find_next_job('App::MtAws::QueueJob::Iterator=HASH(0x16d46c8)') called at lib/App/MtAws/QueueJob/Iterator.pm line 100
    App::MtAws::QueueJob::Iterator::on_itt_and_jobs('App::MtAws::QueueJob::Iterator=HASH(0x16d46c8)') called at lib/App/MtAws/QueueJob.pm line 99
    App::MtAws::QueueJob::next('App::MtAws::QueueJob::Iterator=HASH(0x16d46c8)') called at lib/App/MtAws/QueueEngine.pm line 78
    App::MtAws::QueueEngine::process('App::MtAws::ParentWorker=HASH(0x16da118)', 'App::MtAws::QueueJob::Iterator=HASH(0x16d46c8)') called at lib/App/MtAws/ParentWorker.pm line 83
    App::MtAws::ParentWorker::process_task('App::MtAws::ParentWorker=HASH(0x16da118)', 'App::MtAws::QueueJob::Iterator=HASH(0x16d46c8)', 'App::MtAws::Journal=HASH(0x10d7ee0)') called at lib/App/MtAws/Command/Sync.pm line 203
    App::MtAws::Command::Sync::**ANON**() called at lib/App/MtAws/ForkEngine.pm line 59
    eval {...} called at lib/App/MtAws/ForkEngine.pm line 59
    App::MtAws::ForkEngine::with_forks(1, 'HASH(0x13126c0)', 'CODE(0x131bbd0)') called at lib/App/MtAws/Command/Sync.pm line 208
    App::MtAws::Command::Sync::run('HASH(0x13126c0)', 'App::MtAws::Journal=HASH(0x10d7ee0)') called at lib/App/MtAws.pm line 164
    App::MtAws::process() called at lib/App/MtAws.pm line 117
    eval {...} called at lib/App/MtAws.pm line 117
    App::MtAws::main() called at ./mtglacier.pl line 5

The dryrun however was fine:

$ mtglacier.pl sync --dry-run --config=glacier.cfg --dir=. --new --filter '+sauvegarde-sbackup-pictures_2013-11-16_18.29.31.213311.berseker.ful.tar.gpg.split_\* -**'
MT-AWS-Glacier, Copyright 2012-2013 Victor Efimov http://mt-aws.com/ Version 1.103

Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ak
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_bt
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ay
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_aj
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ar
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ao
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_bi
[...]
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_au
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_bf
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_bk
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_av
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_bn
Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_cb
OK DONE

Am I doing something wrong?

Thanks,
Thibault.

**Comment by vsespb** [comment]
that line
`App::MtAws::Utils::file_size('backup-pictures_...') called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 50`
is strange

i've tried to reproduce bug with same filenames as you have, synced from current directory, but in my case filename is full

`App::MtAws::Utils::file_size("/tmp/UPLFAIL/backup-pictures_2013-11-16_18.29.31.213311.ful.t"...) called at /home/prj/mt/src/lib/App/MtAws/QueueJob/MultipartCreate.pm line 50`

could you please
1. run `mtglacier --version` and post output here (this will list versions of all libraries)?
2. `perl -V` and post here
3. your OS name and version
4. is your current directory a symlink? symlink to symlink? root directory? something else funcy?
5. do you have other directories with symlinks inside current directory?
6. maybe files to upload are symlinks?

**Comment by vsespb** [comment]
> I've tried to reproduce bug with same filenames as you have, synced from current directory, but in my case filename is full

of course I meant that I changed code to return file not found, while file exists. otherwise unaltered code runs without error.

**Comment by vsespb** [comment]
I could imaging something like this happening if you have too strict permissions. example

```
$ mkdir /tmp/TEST1
$ cd /tmp/TEST1
$ pwd
/tmp/TEST1
$ mkdir A
$ cd A
$ mkdir B
$ cd B
$ pwd
/tmp/TEST1/A/B
$ chmod 444 ../..
$ echo Hello > 1
$ cat 1
Hello
$ cat /tmp/TEST1/A/B/1
cat: /tmp/TEST1/A/B/1: Permission denied
$ chmod 777 ../..
$ cat /tmp/TEST1/A/B/1
Hello
```

( we see that filename `1` is accessible, but `/tmp/TEST1/A/B/1` is not, and they both point to same file)

under same circumstances perl code to determine current directory might fail (under some OS).
although I still don't see how this could result in error above.
so in additional to what I asked above, I need output of the following commands inside this directory (under same user as you run mtglacier)
1. `perl -MCwd -e 'print Cwd::getcwd, "\t", $Cwd::VERSION, "\n"'`
2. `perl -MCwd -e 'print Cwd::_perl_abs_path("."), "\t", $Cwd::VERSION, "\n"'`

Also I need permissions of current directory and each parent direcory

example command to diagnose that:

```
$ ls -la .|grep ' \.$'
drwxrwxrwx 3 vse  vse   4096 Dec 15 18:32 .
$ ls -la ..|grep ' \.$'
drwxrwxrwx  3 root root  73728 Dec 15 18:12 .
$ ls -la ../..|grep ' \.$'
drwxrwxrwt 48 root    root      245760 Dec 15 18:42 .
$ ls -la ../../..|grep ' \.$'
```

and I need username under which you run `mtglacier`

**Comment by yo8192** [comment]
Hi there,

I don't have any hard or sym link in there.  The only thing a bit unusual
is that '..' (i.e. the parent dir) is not readable by the user running
mtglacier.pl.  That may be what triggers the exception?

I run mtglacier on Ubuntu Precise (kernel 3.2.0-57-generic).  See the
output of mtglacier --version below.
If you still need it I can send you the additional info you requested later.

Regards,
Thibault.

ps:
$ ./mtglacier.pl --version
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
    LANGUAGE = "en_GB:en",
    LC_ALL = (unset),
    LC_MESSAGES = "fr_FR.UTF-8",
    LC_COLLATE = "fr_FR.UTF-8",
    LC_CTYPE = "fr_FR.UTF-8",
    LANG = "en_GB.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
MT-AWS-Glacier, Copyright 2012-2013 Victor Efimov
http://mt-aws.com/Version 1.103

mt-aws-glacier version: 1.103
Perl Version: 5.014002
AutoLoader    5.71    /usr/share/perl/5.14/AutoLoader.pm
Carp    1.20    /usr/share/perl/5.14/Carp.pm
Carp::Heavy    undef    /usr/share/perl/5.14/Carp/Heavy.pm
Class::Struct    0.63    /usr/share/perl/5.14/Class/Struct.pm
Config    undef    /usr/lib/perl/5.14/Config.pm
Cwd    3.36    /usr/lib/perl/5.14/Cwd.pm
Digest::SHA    5.61    /usr/lib/perl/5.14/Digest/SHA.pm
Digest::base    1.16    /usr/share/perl/5.14/Digest/base.pm
DynaLoader    1.13    /usr/lib/perl/5.14/DynaLoader.pm
Encode    2.42_01    /usr/lib/perl/5.14/Encode.pm
Encode::Alias    2.13    /usr/lib/perl/5.14/Encode/Alias.pm
Encode::Config    2.05    /usr/lib/perl/5.14/Encode/Config.pm
Encode::Encoding    2.05    /usr/lib/perl/5.14/Encode/Encoding.pm
Errno    1.13    /usr/lib/perl/5.14/Errno.pm
Exporter    5.64_03    /usr/share/perl/5.14/Exporter.pm
Exporter::Heavy    5.64_03    /usr/share/perl/5.14/Exporter/Heavy.pm
Fcntl    1.11    /usr/lib/perl/5.14/Fcntl.pm
File::Basename    2.82    /usr/share/perl/5.14/File/Basename.pm
File::Find    1.19    /usr/share/perl/5.14/File/Find.pm
File::Path    2.08_01    /usr/share/perl/5.14/File/Path.pm
File::Spec    3.33    /usr/lib/perl/5.14/File/Spec.pm
File::Spec::Unix    3.33    /usr/lib/perl/5.14/File/Spec/Unix.pm
File::Temp    0.22    /usr/share/perl/5.14/File/Temp.pm
File::stat    1.05    /usr/share/perl/5.14/File/stat.pm
FindBin    1.50    /usr/share/perl/5.14/FindBin.pm
Getopt::Long    2.38    /usr/share/perl/5.14/Getopt/Long.pm
HTTP::Date    6.00    /usr/share/perl5/HTTP/Date.pm
HTTP::Headers    6.00    /usr/share/perl5/HTTP/Headers.pm
HTTP::Message    6.01    /usr/share/perl5/HTTP/Message.pm
HTTP::Request    6.00    /usr/share/perl5/HTTP/Request.pm
HTTP::Response    6.01    /usr/share/perl5/HTTP/Response.pm
HTTP::Status    6.00    /usr/share/perl5/HTTP/Status.pm
I18N::Langinfo    0.08    /usr/lib/perl/5.14/I18N/Langinfo.pm
IO    1.25_04    /usr/lib/perl/5.14/IO.pm
IO::Handle    1.31    /usr/lib/perl/5.14/IO/Handle.pm
IO::Pipe    1.14    /usr/lib/perl/5.14/IO/Pipe.pm
IO::Seekable    1.1    /usr/lib/perl/5.14/IO/Seekable.pm
IO::Select    1.20    /usr/lib/perl/5.14/IO/Select.pm
JSON::XS    2.32    /usr/lib/perl5/JSON/XS.pm
LWP    6.03    /usr/share/perl5/LWP.pm
LWP::MemberMixin    undef    /usr/share/perl5/LWP/MemberMixin.pm
LWP::Protocol    6.00    /usr/share/perl5/LWP/Protocol.pm
LWP::UserAgent    6.03    /usr/share/perl5/LWP/UserAgent.pm
List::Util    1.23    /usr/lib/perl/5.14/List/Util.pm
MIME::Base64    3.13    /usr/lib/perl/5.14/MIME/Base64.pm
POSIX    1.24    /usr/lib/perl/5.14/POSIX.pm
PerlIO::encoding    0.14    /usr/lib/perl/5.14/PerlIO/encoding.pm
Scalar::Util    1.23    /usr/lib/perl/5.14/Scalar/Util.pm
SelectSaver    1.02    /usr/share/perl/5.14/SelectSaver.pm
Storable    2.27    /usr/lib/perl/5.14/Storable.pm
Symbol    1.07    /usr/share/perl/5.14/Symbol.pm
Tie::Hash    1.04    /usr/share/perl/5.14/Tie/Hash.pm
Time::Local    1.2000    /usr/share/perl/5.14/Time/Local.pm
URI    1.59    /usr/share/perl5/URI.pm
URI::Escape    3.31    /usr/share/perl5/URI/Escape.pm
XSLoader    0.13    /usr/share/perl/5.14/XSLoader.pm
attributes    0.14    /usr/lib/perl/5.14/attributes.pm
base    2.16    /usr/share/perl/5.14/base.pm
bytes    1.04    /usr/share/perl/5.14/bytes.pm
common::sense    3.4    /usr/share/perl5/common/sense.pm
constant    1.21    /usr/share/perl/5.14/constant.pm
integer    1.00    /usr/share/perl/5.14/integer.pm
lib    0.63    /usr/lib/perl/5.14/lib.pm
overload    1.13    /usr/share/perl/5.14/overload.pm
strict    1.04    /usr/share/perl/5.14/strict.pm
utf8    1.09    /usr/share/perl/5.14/utf8.pm
vars    1.02    /usr/share/perl/5.14/vars.pm
warnings    1.12    /usr/share/perl/5.14/warnings.pm
warnings::register    1.02    /usr/share/perl/5.14/warnings/register.pm
OK DONE

2013/12/15 Victor Efimov notifications@github.com

**Comment by vsespb** [comment]
I have same ubuntu version and same perl, but can reproduce only part of problem (i.e. when parent directory is unreadable - files in it are unreadable by _absolute paths_ - it will be unreadable by any software). so need more info.

**Comment by yo8192** [comment]
Hi there,

Indeed the problem is with the parent dir which is not readable and not executable by the user used to run the script.

I run the script as the user 'nobody', so I ran the commands you gave as 'nobody' too:

$ sudo -u nobody perl -MCwd -e 'print Cwd::getcwd, "\t", $Cwd::VERSION, "\n"'
/data/backup/thib/amazon-glacier    3.36
$ sudo -u nobody perl -MCwd -e 'print Cwd::_perl_abs_path("."), "\t", $Cwd::VERSION, "\n"'
Can't locate object method "rel2abs" via package "File::Spec" at /usr/lib/perl/5.14/Cwd.pm line 412.
$ ls -la .|grep ' .$'
drwxr-xr-x  4 john   john       12288 Dec 15 11:09 .
$ ls -la ..|grep ' .$'
drwxr-x--- 18 john john        4096 Dec 15 17:27 .
$ ls -la ../..|grep ' .$'
drwxr-xr-x  5 john john  4096 Dec 13 17:41 .
$ ls -la ../../..|grep ' .$'
drwxrwxrwx  8 root root      4096 Nov 16 09:17 .

With that config, I get the error I reported.

Now if I run:
$ chmod o+rx ..

Then it works fine, no exception anymore.

So do you think the problem is that it tries to resolve to absolute paths even though I only specified relative ones?

Regards,
Thibault.

**Comment by vsespb** [comment]
> So do you think the problem is that it tries to resolve to absolute paths even though I only specified relative ones?

such problem exists - it was intentional that `mtglacier` internally operate absolute paths. now I think it's perhaps wrong.

however you seem to observe different problem.

I reproduced your filesystem setup (and all modules, exactly same versions on clean ubuntu machine), and I am getting

```
App::MtAws::Utils::file_size('/tmp/TEST/A/1') called at /usr/share/perl5/App/MtAws/QueueJob/MultipartCreate.pm line 50
```

(absolute filename)
but in your case

```
App::MtAws::Utils::file_size('backup-pictures_...') called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 50
```

(relative filename).

There could be bugs in get-current-directory perl function which lead to similar results (in case of permissions problems), but strange that I could not reproduce this on exactly same, clean system.

**Comment by vsespb** [comment]
Is it i386 or x86_64?
Is some part of `/data/backup/thib/amazon-glacier` a symlink/bind directory/mount point?
What filesystem? NFS/AFS? EXT4?

Could you try to reproduce in different directory (with same permissions setup), say, in `/tmp` ?

( i.e. enough to reproduce problem with `--dry-run` because
that line
`Will UPLOAD backup-pictures_2013-11-16_18.29.` already shows problem - there should be absolute filename, not relative)

**Comment by vsespb** [comment]
> (please note I've shortened the file names in the outputs below)

oh, just noticed that.

did you shortened this line

`App::MtAws::Utils::file_size('backup-pictures_...') called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 50`

or this

`Will UPLOAD backup-pictures_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ak`

or both? chance that there was a full path?

**Comment by yo8192** [comment]
Hi there,

x86_64, and yes, /data is a ext4 partition.
It's possible I removed the full paths and made it look like relative
paths, I honestly can't remember.  Apolgies if that created confusion.

However I can reproduce it in /tmp too.  This time I used fake files which
already have a shorter name -- I didn't amend the file names in the below:

/tmp/backup/thib/amazon-glacier$ sudo -u nobody perl -Ilib
./mtglacier.plsync --dry-run  --config=glacier.cfg --dir=. --new
--filter
'+backup-pictures2_2013-11-16_18.29.31.213311.ful.tar.gpg.split_\* -**'
MT-AWS-Glacier, Copyright 2012-2013 Victor Efimov
http://mt-aws.com/Version 1.103

Will UPLOAD
/tmp/backup/thib/amazon-glacier/backup-pictures2_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ak
OK DONE

$ sudo -u nobody perl -Ilib ./mtglacier.pl sync   --config=glacier.cfg
--dir=. --new --filter
'+backup-pictures2_2013-11-16_18.29.31.213311.ful.tar.gpg.split_\* -**'
MT-AWS-Glacier, Copyright 2012-2013 Victor Efimov
http://mt-aws.com/Version 1.103

PID 12437 Started worker
PID 12438 Started worker
PID 12439 Started worker
PID 12440 Started worker
UNEXPECTED ERROR (parent): file not exists at lib/App/MtAws/Utils.pm line
160

App::MtAws::Utils::file_size('/tmp/backup/thib/amazon-glacier/backup-pictures2_2013-11-16_1...')
called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 50

App::MtAws::QueueJob::MultipartCreate::init_file('App::MtAws::QueueJob::MultipartCreate=HASH(0x2195298)')
called at lib/App/MtAws/QueueJob/MultipartCreate.pm line 72

App::MtAws::QueueJob::MultipartCreate::on_create('App::MtAws::QueueJob::MultipartCreate=HASH(0x2195298)')
called at lib/App/MtAws/QueueJob.pm line 99

App::MtAws::QueueJob::next('App::MtAws::QueueJob::MultipartCreate=HASH(0x2195298)')
called at lib/App/MtAws/QueueJob.pm line 88

App::MtAws::QueueJob::next('App::MtAws::QueueJob::UploadMultipart=HASH(0x219a8d0)')
called at lib/App/MtAws/QueueJob.pm line 88

App::MtAws::QueueJob::next('App::MtAws::QueueJob::Upload=HASH(0x21a0530)')
called at lib/App/MtAws/QueueJob/Iterator.pm line 61

App::MtAws::QueueJob::Iterator::find_next_job('App::MtAws::QueueJob::Iterator=HASH(0x1995460)')
called at lib/App/MtAws/QueueJob/Iterator.pm line 100

App::MtAws::QueueJob::Iterator::on_itt_and_jobs('App::MtAws::QueueJob::Iterator=HASH(0x1995460)')
called at lib/App/MtAws/QueueJob.pm line 99

App::MtAws::QueueJob::next('App::MtAws::QueueJob::Iterator=HASH(0x1995460)')
called at lib/App/MtAws/QueueJob/Iterator.pm line 61

App::MtAws::QueueJob::Iterator::find_next_job('App::MtAws::QueueJob::Iterator=HASH(0x21a0398)')
called at lib/App/MtAws/QueueJob/Iterator.pm line 100

App::MtAws::QueueJob::Iterator::on_itt_and_jobs('App::MtAws::QueueJob::Iterator=HASH(0x21a0398)')
called at lib/App/MtAws/QueueJob.pm line 99

App::MtAws::QueueJob::next('App::MtAws::QueueJob::Iterator=HASH(0x21a0398)')
called at lib/App/MtAws/QueueEngine.pm line 78

App::MtAws::QueueEngine::process('App::MtAws::ParentWorker=HASH(0x21a08a8)',
'App::MtAws::QueueJob::Iterator=HASH(0x21a0398)') called at
lib/App/MtAws/ParentWorker.pm line 83

App::MtAws::ParentWorker::process_task('App::MtAws::ParentWorker=HASH(0x21a08a8)',
'App::MtAws::QueueJob::Iterator=HASH(0x21a0398)',
'App::MtAws::Journal=HASH(0x1b9e7d0)') called at
lib/App/MtAws/Command/Sync.pm line 203
        App::MtAws::Command::Sync::**ANON**() called at
lib/App/MtAws/ForkEngine.pm line 59
        eval {...} called at lib/App/MtAws/ForkEngine.pm line 59
        App::MtAws::ForkEngine::with_forks(1, 'HASH(0x1dd8f70)',
'CODE(0x1de2460)') called at lib/App/MtAws/Command/Sync.pm line 208
        App::MtAws::Command::Sync::run('HASH(0x1dd8f70)',
'App::MtAws::Journal=HASH(0x1b9e7d0)') called at lib/App/MtAws.pm line 164
        App::MtAws::process() called at lib/App/MtAws.pm line 117
        eval {...} called at lib/App/MtAws.pm line 117
        App::MtAws::main() called at ./mtglacier.pl line 5

$ cd /tmp ; find backup/ -ls | grep -v 'lib/'
1000120    4 drwxr-xr-x   3 john     john     4096 Dec 15 22:12 backup/
1000121    4 drwxr-x---   3 john     john     4096 Dec 15 22:12 backup/thib
1057840   12 drwxr-xr-x   4 john     john    12288 Dec 15 22:28
backup/thib/amazon-glacier
1063888    4 -rw-r--r--   1 john     john       85 Dec 15 22:20
backup/thib/amazon-glacier/backup-pictures2_2013-11-16_18.29.31.213311.ful.tar.gpg.split_ak
1061607    4 drwxr-xr-x   3 john     john     4096 Dec 14 18:27
backup/thib/amazon-glacier/lib
1063886    4 drwxr-xr-x   2 nobody   nogroup      4096 Dec 15 22:18
backup/thib/amazon-glacier/journal
1063887    0 -rw-r--r--   1 nobody   nogroup         0 Dec 15 22:18
backup/thib/amazon-glacier/journal/journal.log
1061604    4 -rwxr-xr-x   1 john     john      103 Dec 14 18:28
backup/thib/amazon-glacier/mtglacier.pl
1061603    4 -rw-r--r--   1 john     john      144 Dec 15 22:28
backup/thib/amazon-glacier/glacier.cfg

Hopefully that will make things clearer and will allow you to reproduce
this.

It looks like the only permission I need to make this work is "chmod o+x
/tmp/backup/thib" so as iit can traverse it.

Regards,
Thibault.

2013/12/15 Victor Efimov notifications@github.com

**Comment by vsespb** [comment]
ok, cool. so now it's absolute filenames, not relative.

`Will UPLOAD /tmp/backup/thib/amazon-glacier/backup-pictures2_2013-11-1`

thus the only issue here we're having is unaccessible parent directory which causes problems because `mtglacier` uses absolute filenames internally.

I will try to fix it, will use relative filenames only.

Will let you know here when fixed. Until this fixed you'll have to wrokaround by adjusting permissions.

And thanks for reporting this.

**Comment by vsespb** [comment]
Hello. Should be fixed in `1.110` - details in ChangeLog. Let me know if problem re-appears. Thanks.
